@xen-orchestra/lite/src/App/TreeView.tsx

```diff
@@ -108,7 +127,9 @@ const TreeView = withState<State, Props, Effects, Computed, ParentState, ParentE
         return collection
       },
       hostsByPool: state => state.objectsByType?.get('host')?.groupBy((host: Host) => host.$pool.$id),
+      objectId: (_, { location }) => location.pathname.split('/')[3],
       pools: state => state.objectsByType?.get('pool'),
+      selectedNodes: ({ objectId, _selectedNodes }) => (objectId !== undefined ? _selectedNodes : []),
```

Comment:
What about relying 100% on the URL? ie returning `objectId !== undefined ? [objectId] : []`.